### PR TITLE
Return `IHtmlContent` from `GetHtml()`

### PR DESCRIPTION
**Pull request by dougbu**
- part of aspnet/Mvc#3123
- also ignore `launchSettings.json` files

### CODE REVIEW

File: src/Microsoft.AspNet.Antiforgery/DefaultAntiforgery.cs [hunk 1]
```diff
@@ -26,34 +25,40 @@ public class DefaultAntiforgery : IAntiforgery
...
             _options = antiforgeryOptionsAccessor.Value;
             _tokenGenerator = tokenGenerator;
             _tokenSerializer = tokenSerializer;
             _tokenStore = tokenStore;
-            _htmlEncoder = htmlEncoder;
         }
 
         /// <inheritdoc />
-        public string GetHtml(HttpContext context)
+        public void GetHtml(HttpContext context, IHtmlContentBuilder content)
```
Comment (rynowak) [inline src/Microsoft.AspNet.Antiforgery/DefaultAntiforgery.cs]: Yeah, having looked at the MVC reaction PR, I really don't get why this is better than just writing a private class here and returning it.

File: test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs [hunk 2]
```diff
@@ -159,15 +161,23 @@ public void GetHtml_ExistingInvalidCookieToken_GeneratesANewCookieAndAnAntiforge
...
-            Assert.Equal(
-                @"<input name=""HtmlEncode[[form-field-name]]"" type=""HtmlEncode[[hidden]]"" " +
-                @"value=""HtmlEncode[[serialized-form-token]]"" />",
-                inputElement);
+            using (var writer = new StringWriter())
+            {
+                content.WriteTo(writer, encoder);
+
+                Assert.Equal(
+                    @"<input name=""HtmlEncode[[form-field-name]]"" type=""hidden"" " +
```
Comment (rynowak) [inline test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs]: Oh noes mah 'hidden' isn't getting encoded :trollface:

File: src/Microsoft.AspNet.Antiforgery/project.json [hunk 3]
```diff
@@ -11,12 +11,12 @@
   },
   "dependencies": {
     "Microsoft.AspNet.DataProtection": "1.0.0-*",
+    "Microsoft.AspNet.Html.Abstractions": "1.0.0-*",
     "Microsoft.AspNet.Http.Abstractions": "1.0.0-*",
-    "Microsoft.AspNet.WebUtilities": "1.0.0-*",
-    "Microsoft.Extensions.WebEncoders": "1.0.0-*"
```
Comment (rynowak) [inline src/Microsoft.AspNet.Antiforgery/project.json]: nice

File: src/Microsoft.AspNet.Antiforgery/DefaultAntiforgery.cs (hunk 1 above)
Comment (dougbu) [inline src/Microsoft.AspNet.Antiforgery/DefaultAntiforgery.cs]: As discussed in aspnet/Mvc#3635, I'll make the switch.

**Comment by dougbu** [comment]
:up::date: and changed title

File: .gitignore [hunk 4]
```diff
@@ -25,4 +25,5 @@ project.lock.json
 *DS_Store
 *.ncrunchsolution
 *.*sdf
-*.ipch
\ No newline at end of file
+*.ipch
+launchSettings.json
```
Comment (pranavkm) [inline .gitignore]: So annoying.

**Comment by pranavkm** [comment]
:shipit:

**Comment by dougbu** [comment]
- 78face4 for `.gitignore` update
- 6a9b38d

**Pull request closed without merging**